testsuite/MDAnalysisTests/coordinates/test_h5md.py

```diff
@@ -0,0 +1,496 @@
+                        decimal=6)
+    # second timestep
+    h5md_universe.trajectory[1]
+    assert_almost_equal(h5md_universe.trajectory.ts.dimensions,
+                        [52.807877, 52.807877, 52.807877, 90., 90., 90.],
+                        decimal=6)
+    # third timestep
+    h5md_universe.trajectory[2]
+    assert_almost_equal(h5md_universe.trajectory.ts.dimensions,
+                        [52.839806, 52.839806, 52.839806, 90., 90., 90.],
+                        decimal=6)
+
+
+@pytest.mark.skipif(not HAS_H5PY, reason="h5py not installed")
+def test_h5md_data_step(h5md_universe):
+    h5md_universe.trajectory[0]
+    assert h5md_universe.trajectory.ts.data['step'] == 0
+    h5md_universe.trajectory[1]
+    assert h5md_universe.trajectory.ts.data['step'] == 25000
+    h5md_universe.trajectory[2]
+    assert h5md_universe.trajectory.ts.data['step'] == 50000
+
+
+@pytest.mark.skipif(not HAS_H5PY, reason="h5py not installed")
+def test_rewind(h5md_universe):
+    h5md_universe.trajectory.rewind()
+    assert h5md_universe.trajectory.ts.frame == 0
+
+
+@pytest.mark.skipif(not HAS_H5PY, reason="h5py not installed")
+def test_next(h5md_universe):
+    h5md_universe.trajectory.rewind()
+    h5md_universe.trajectory.next()
+    assert h5md_universe.trajectory.ts.frame == 1
+
+
+@pytest.mark.skipif(not HAS_H5PY, reason="h5py not installed")
+def test_jump_last_frame(h5md_universe):
+    h5md_universe.trajectory[-1]
+    assert h5md_universe.trajectory.ts.frame == 2
+
+
+@pytest.mark.skipif(not HAS_H5PY, reason="h5py not installed")
+@pytest.mark.parametrize("start, stop, step", ((0, 2, 1),
+                                               (1, 2, 1)))
+def test_slice(h5md_universe, start, stop, step):
+    frames = [h5md_universe.trajectory.ts.frame
+              for ts in h5md_universe.trajectory[start:stop:step]]
+    assert_array_equal(frames, np.arange(start, stop, step))
+
+
+@pytest.mark.skipif(not HAS_H5PY, reason="h5py not installed")
+@pytest.mark.parametrize("array_like", [list, np.array])
+def test_array_like(h5md_universe, array_like):
+    array = array_like([0, 2])
+    frames = [h5md_universe.trajectory.ts.frame
+              for ts in h5md_universe.trajectory[array]]
+    assert_array_equal(frames, array)
+
+
+@pytest.mark.skipif(not HAS_H5PY, reason="h5py not installed")
+@pytest.mark.parametrize("indices", ([0, 1, 2, 1, 2, 2, 0]))
+def test_list_indices(h5md_universe, indices):
+    frames = [h5md_universe.trajectory.ts.frame
+              for ts in h5md_universe.trajectory[indices]]
+    assert_array_equal(frames, indices)
+
+
+@pytest.fixture
+@pytest.mark.skipif(not HAS_H5PY, reason="h5py not installed")
+def ref():
+    return H5MDReference()
+
+
+@pytest.fixture
+@pytest.mark.skipif(not HAS_H5PY, reason="h5py not installed")
+def h5md_file():
+    return h5py.File(H5MD_xvf, 'r')
+
+
+@pytest.mark.skipif(not HAS_H5PY, reason="h5py not installed")
+def test_has_position(h5md_file, ref, tmpdir):
```

Comment:
This test does not assert anything – add some sort of assertion.

Reply:
I have 2 ideas, which is better:

```
assert getattr(u.trajectory.ts, 'positions', None) is None
```
```
with pytest raises (NoDataError):
    u.trajectory.ts.positions
```

Reply:
Also, come to think of it, the test should probably be called `test_no_positions`...